spring-kafka/src/main/java/org/springframework/kafka/security/jaas/JaasInitializerListener.java

```diff
@@ -0,0 +1,117 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+
+import org.apache.kafka.common.security.JaasUtils;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.util.Assert;
+
+/**
+ * Configures JAAS.
+ *
+ * @author Marius Bogoevici
+ * @author Gary Russell
+ *
+ * @since 2.0
+ */
+public class JaasInitializerListener
+		implements ApplicationListener<ContextRefreshedEvent>, ApplicationContextAware, DisposableBean {
+
+	private ApplicationContext applicationContext;
+
+	private final boolean ignoreJavaLoginConfigParamSystemProperty;
+
+	private final File placeholderJaasConfiguration;
+
+	private final JaasLoginModuleConfiguration jaasLoginModuleConfiguration;
+
+	public JaasInitializerListener(JaasLoginModuleConfiguration jaasLoginModuleConfiguration) throws IOException {
+		// we ignore the system property if it wasn't originally set at launch
+		this.ignoreJavaLoginConfigParamSystemProperty = (System.getProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM) == null);
+		this.placeholderJaasConfiguration = File.createTempFile("kafka-client-jaas-config-placeholder", "conf");
+		this.placeholderJaasConfiguration.deleteOnExit();
+		this.jaasLoginModuleConfiguration = jaasLoginModuleConfiguration;
+	}
+
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+		this.applicationContext = applicationContext;
+	}
+
+	@Override
+	public void destroy() throws Exception {
+		if (this.ignoreJavaLoginConfigParamSystemProperty) {
+			System.clearProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM);
+		}
+	}
+
+	@Override
+	public void onApplicationEvent(ContextRefreshedEvent event) {
+		if (event.getSource() == this.applicationContext) {
+			// only use programmatic support if a file is not set via system property
+			if (this.ignoreJavaLoginConfigParamSystemProperty && this.jaasLoginModuleConfiguration != null) {
+				Map<String, AppConfigurationEntry[]> configurationEntries = new HashMap<>();
+				AppConfigurationEntry kafkaClientConfigurationEntry = new AppConfigurationEntry(
+						this.jaasLoginModuleConfiguration.getLoginModule(),
+						this.jaasLoginModuleConfiguration.getControlFlagValue(),
+						this.jaasLoginModuleConfiguration.getOptions());
+				configurationEntries.put(JaasUtils.LOGIN_CONTEXT_CLIENT,
+						new AppConfigurationEntry[] { kafkaClientConfigurationEntry });
+				Configuration.setConfiguration(new InternalConfiguration(configurationEntries));
+				// Workaround for a 0.9 client issue where even if the Configuration is
```

Comment:
Maybe we don't need this workaround anymore since we are far away from `0.9` Client?
OTOH how about to switch to the `SmartInitializingSingleton` instead of `ApplicationListener` and rename the class to the `KafkaJaasInitializer`. Although, IMHO, we don't need this class at all and its functionality can be covered by the `JaasLoginModuleConfiguration`, but may be with more Kafka-specific name...